### ISSUE TITLE
Provision for setting default value must be present in Column class

### ISSUE THREAD

**Issue by pankajgoyalhitk** [phalcon/cphalcon#10996]
In a Column Object, we can mention whether the field is null or notNull. But if the notNull property is false, we do not have any scope of mentioning its default value.

**Comment by pankajgoyalhitk** [comment]
Copied from here : https://docs.phalconphp.com/en/latest/reference/db.html#creating-tables

$connection->createTable(
    "robots",
    null,
    array(
       "columns" => array(
            new Column(
                "id",
                array(
                    "type"          => Column::TYPE_INTEGER,
                    "size"          => 10,
                    "notNull"       => true,
                    "autoIncrement" => true,
                    "primary"       => true,
                )
            ),
            new Column(
                "name",
                array(
                    "type"    => Column::TYPE_VARCHAR,
                    "size"    => 70,
                    "notNull" => true,
                )
            ),
            new Column(
                "year",
                array(
                    "type"    => Column::TYPE_INTEGER,
                    "size"    => 11,
                    "notNull" => false,
                    "default" => 2015
                )
            )
        )
    )
);

In the last column object, I set notNull to false and want to give a default column value (like: "default" => 2015). But support for providing default value in Column Object itself is not present. It will by default take the default value as NULL.

**Comment by sergeyklay** [comment]
```php
$connection->createTable(
    'robots',
    null,
    [
        'columns' => [
              new Column('year', [
                'type'    => Column::TYPE_INTEGER,
                'size'    => 11,
                'notNull' => true,
                'default' => 2015,
            ])
        ]
    ]
);
```

**Comment by pankajgoyalhitk** [comment]
I tried it but it didn't work. Have you tried it? then, can you please say me the phalcon version you are using?

**Comment by sergeyklay** [comment]
```
mysql> show create table users\G
*************************** 1. row ***************************
       Table: users
Create Table: CREATE TABLE `users` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `username` varchar(32) COLLATE utf8_unicode_ci NOT NULL,
  `password` char(40) COLLATE utf8_unicode_ci NOT NULL,
  `name` varchar(120) COLLATE utf8_unicode_ci NOT NULL,
  `email` varchar(70) COLLATE utf8_unicode_ci NOT NULL,
  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `active` char(1) COLLATE utf8_unicode_ci NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci
1 row in set (0.00 sec)
```

```php
    public function morph()
    {
        $this->morphTable('users', array(
                'columns' => array(
                    new Column(
                        'id',
                        array(
                            'type' => Column::TYPE_INTEGER,
                            'unsigned' => true,
                            'notNull' => true,
                            'autoIncrement' => true,
                            'size' => 10,
                            'first' => true
                        )
                    ),
                    new Column(
                        'username',
                        array(
                            'type' => Column::TYPE_VARCHAR,
                            'notNull' => true,
                            'size' => 32,
                            'after' => 'id'
                        )
                    ),
                    new Column(
                        'password',
                        array(
                            'type' => Column::TYPE_CHAR,
                            'notNull' => true,
                            'size' => 40,
                            'after' => 'username'
                        )
                    ),
                    new Column(
                        'name',
                        array(
                            'type' => Column::TYPE_VARCHAR,
                            'notNull' => true,
                            'size' => 120,
                            'after' => 'password'
                        )
                    ),
                    new Column(
                        'email',
                        array(
                            'type' => Column::TYPE_VARCHAR,
                            'notNull' => true,
                            'size' => 70,
                            'after' => 'name'
                        )
                    ),
                    new Column(
                        'created_at',
                        array(
                            'type' => Column::TYPE_TIMESTAMP,
                            'default' => "CURRENT_TIMESTAMP",
                            'notNull' => true,
                            'size' => 1,
                            'after' => 'email'
                        )
                    ),
                    new Column(
                        'active',
                        array(
                            'type' => Column::TYPE_CHAR,
                            'notNull' => true,
                            'default' => 'Y',
                            'size' => 1,
                            'after' => 'created_at'
                        )
                    )
                ),
                'indexes' => array(
                    new Index('PRIMARY', array('id'), 'PRIMARY')
                ),
                'options' => array(
                    'TABLE_TYPE' => 'BASE TABLE',
                    'AUTO_INCREMENT' => '2',
                    'ENGINE' => 'InnoDB',
                    'TABLE_COLLATION' => 'utf8_unicode_ci'
                ),
            )
        );
    }
```

```
mysql> show create table users\G
*************************** 1. row ***************************
       Table: users
Create Table: CREATE TABLE `users` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `username` varchar(32) COLLATE utf8_unicode_ci NOT NULL,
  `password` char(40) COLLATE utf8_unicode_ci NOT NULL,
  `name` varchar(120) COLLATE utf8_unicode_ci NOT NULL,
  `email` varchar(70) COLLATE utf8_unicode_ci NOT NULL,
  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `active` char(1) COLLATE utf8_unicode_ci NOT NULL DEFAULT 'Y',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci

```

**Comment by sergeyklay** [comment]
$ php --ri phalcon | grep Version
Version => 2.0.9
Powered by Zephir => Version 0.9a-dev

**Comment by sergeyklay** [comment]
@SidRoberts Can you please update doc?

**Comment by SidRoberts** [comment]
Fixed in phalcon/docs@ab85fa10a176234fb6a3f42dd71e54f9a031d66e.

**Comment by pankajgoyalhitk** [comment]
Thank you very much !!
I am using 2.0.5 version - that's why I am facing this issue.
php --ri phalcon | grep Version
Version => 2.0.5
Powered by Zephir => Version 0.7.1b
